pkg/user/signer.go

```
@@ -153,7 +153,28 @@ func (s *Signer) AccountByAddress(address sdktypes.AccAddress) *Account {
 
 	accountName, exists := s.addressToAccountMap[addrStr]
 	if !exists {
-		return nil
+		record, err := s.keys.KeyByAddress(address)
+		if err != nil {
+			return nil
+		}
+
+		acc := &Account{
+			name:          record.Name,
+			address:       address,
+			accountNumber: 0,
+			sequence:      0,
+		}
+
+		pk, err := record.GetPubKey()
+		if err != nil {
+			return nil
```

[blocking][question] why does this silently ignore an error?

---

> [blocking][question] why does this silently ignore an error?

@avorylli 

I think it doesn't silently ignore the error. It intentionally returns nil when it can't get the public key, as this indicates we can't fully populate the account.

---

It is confusing to me that this function can return `nil` instead of a valid address especially because that's not communicated in the function signature. I would've expected

```
func (s *Signer) AccountByAddress(address sdktypes.AccAddress) *Account {
```

to always return a valid account. If it is possible for this function to _not_ return an account then the function signature should be modified to

```
func (s *Signer) AccountByAddress(address sdktypes.AccAddress) (*Account, error) {
```

so that if `nil` is returned for the account, `err` can contain some information on _why_ the account was not returned.

---

not having a public key is an error imo as well @avorylli. we should return an error